src/main/controllers/UserAccountController.kt

```diff
@@ -37,7 +37,7 @@ class UserAccountController: DefaultController<UserAccount>(), RestController<Us
 
         val apiCred = user.apiCreds
 
-        ValidateApiKeyService.execute(apiCred.apiKey, apiCred.secretKey)
+        ValidateApiKeyService.execute(apiCred.apiKey, request.input["secretKey"] as String)
```

Comment:
maybe we should refactor ValidateApiKeyService to take a user and secret -- that way it wont have to query for apicred again -- ill make this change just make sure to rebase